docs/Conceptual/00 - Introduction.md

```diff
@@ -0,0 +1,200 @@
+
+That solution is likely implemented using many compute and storage resources, distributed across various physical and virtual machines, but it is conceptually *one solution, lots of users*. To developer and user alike, it looks like a single set of resources, serving many users. Those users may work together in logical groups (like teams, or organizations) but they are fundamentally sharing a single solution.
+
+Multi-tenancy, on the other hand, can be thought of as *lots of solutions, lots of users*. To the *owner* of the multi-tenanted solution provider, it looks conceptually like *many isolated child solutions* (or *tenants*) consuming the resources it provides, each of which independently serves many users. To any individual tenant, their "tenancy" looks like a single set of resources that they use to serve their users, and no-one else’s.
+
+![multi-user v. multi tenant](multi-user v multi-tenant.png)
+
+This *conceptual difference* is often confused with the *implementation difference*. It is perfectly possible for the provider of a multi-tenanted service to implement their service using a single set of resources shared between all their tenants (if other constraints permit).
+
+## When should a solution be multi-tenanted, not multi-user?
+A multi-tenanted solution is required when one or more groups of users have a strong requirement for *isolation* from other groups of users. This is not simply a secure connection to services for that tenant. Isolation will typically manifest in one or more of the following ways:
+
+1. The ability for a tenant to provision, manage, and monitor a collection of resources independently of other tenants.
+1. The ability for a tenant to bring their own resources (including, for example, domain names, application code, configuration, data, cryptographic keys, policy) and integrate into their solution
+1. The ability to provide particular kinds of security boundary around their solution and control identity and access (including the ability to deny access to the entity that provides the tenant with those resources)
+1. The inability for one tenant to determine the existence or otherwise of another tenant.
+1. The inability for one tenant to interfere with the operation or performance of another tenant e.g. by starving it of resources.
+
+Note that this is not simply *data isolation*. Most data storage solutions (blob storage, SQL databases etc.) have mechanisms that allow you to control access to subsets of the data, constrained to particular identities. For example, techniques like Row Level Security (RLS), Attribute Based Access Control (ABAC), filtering, and data partitioning provide various types of security and access boundary. Multi-user systems will often use these techniques to ensure segregation of user data.
+
+A multi-tenanted system will also use these techniques to provide data segregation, but adds in other constraints from the list above. 
+
+## Tenancy as a hierarchy
```

Comment:
I think one of the important and non-obvious points of this section is the heterogenous nature of the hierarchy. When I see "Tenancy as a hierarchy" I leap to the composite pattern, in which all nodes are potential parents of other similar nodes, and in similar ways. But that's the wrong mental model, because in the case of Azure, it's not a self-similar tree of limitless depth. The nature of the nodes is quite different at every level.

Reply:
That's a very good point. I'll try to draw that out more clearly.